### PR TITLE
Transients

### PR DESCRIPTION
**Pull request by sybenzvi**
This very long-lived branch should be ready to merge.

Note for reviewers: the function `desisim.io.empty_metatable` now inserts 4 columns for transient metadata into the objmeta/objtruth tables, replacing a separate metatable used only for transients. The columns are present whether or not a transient is simulated. We could clean that up but it would also require changes to desitarget and I wanted to avoid forcing a coordinated merge of both projects.

### CODE REVIEW

File: py/desisim/io.py [hunk 1]
```diff
@@ -1172,6 +1172,12 @@ def empty_metatable(nmodel=1, objtype='ELG', subtype='', simqso=False, input_met
                                   data=np.zeros(nmodel)-1, unit='Dex'))
         objmeta.add_column(Column(name='SIIHBETA', length=nmodel, dtype='f4',
                                   data=np.zeros(nmodel)-1, unit='Dex'))
+        objmeta.add_column(Column(name='TRANSIENT_MODEL', length=nmodel, dtype='U20'))
+        objmeta.add_column(Column(name='TRANSIENT_TYPE', length=nmodel, dtype='U10'))
+        objmeta.add_column(Column(name='TRANSIENT_EPOCH', length=nmodel, dtype='f4',
+                                  data=np.zeros(nmodel)-1, unit='day'))
+        objmeta.add_column(Column(name='TRANSIENT_RFLUXRATIO', length=nmodel, dtype='f4',
+                                  data=np.zeros(nmodel)-1))
```
[the remark this answers is not in the document]
Comment (moustakas) [reply py/desisim/io.py:1181]: @sybenzvi I see now that this is exactly the comment you made when you opened this PR! Let's merge this PR (once tests pass) but could you please open a ticket to only add the `TRANSIENT_` columns if one is simulating transients (even if it requires coordination with a `desitarget` PR, just to we don't lose track?

**Comment by moustakas** [comment]
@sbailey, the test failures are totally unrelated to this PR. I tried testing against `desispec/0.33.0` rather than `desisipec/master` but that wasn't a quick fix. Could you take a quick peek?  Here's the main error:
```
File "/home/travis/miniconda/envs/test/lib/python3.5/site-packages/desispec/io/fibermap.py", line 324

    expdir = f'{nightdir}/{expid:08d}'
```

**Comment by sbailey** [comment]
desispec now requires python 3.6, which broke the desisim python 3.5 tests.  I dropped those tests.

**Comment by sbailey** [comment]
after some tweaking to the travis config, tests pass there again.  merging

**Pull request merged**